### DISCUSSION TITLE
Pinecil build from source

### DISCUSSION THREAD

**Discussion by Mjolinor** [Ralim/IronOS#832] [General]
I bought two Pinecil irons. They both worked briefly but packed up after five minutes or so. The problem being the accelerometer chip that seems particularly fragile. I removed the blown chip but cannot see how replacing it will do any good. Pinecil tell me that anything over 3.8 volts will blow the BMA223 and the oscilloscope shows me a 30 volt spike on the I2C when you plug USB or DC power into it. Even in the steady state the I2C signals overshoot to 4.3 volts.

I figure that the easiest solution should be to compile with no accelerometer support. I have no problem not having that particular feature. I had hoped that the firmware would still work when it did not get anything back from the BMA223 but it still has no display though it does now go into DFU mode OK.

Can anyone start me off by pointing me in the right direction?

**Comment by Ralim** [comment]
This is interesting that you are seeing a spike on the i2c; Had a quick probe of my disassembled unit here and do not see anything like that on it.

Are you sure your power supplies are grounded ?
Have had one unit damaged when my power supply was floating at like 100V ac above ground (bad isolation filter caps).

The firmware will operate fine without the accelerometer. On the latest build it will give you a warning on the first two power ups, but it will continue to operate and will just lock out all accelerometer based options to being turned off.


If you are seeing nothing on the display once its been flashed it _sounds like_ whatever is causing your issue at turn on has also affected your OLED display as that is also on the I2C bus.

**Reply by Mjolinor** [reply]
Thanks. It seems that both are dead then. Pinecil refunded me for both irons but I would rather have solved the problem than got a refund. Some images here of the spikes I am seeing and the supplies I used to power the iron. http://rollingpast.com/john/Pinecil/

I don't know why no other irons have failed, only my two but for sure there is nothing wrong with my bench supplies though I cannot comment on the USB chargers that I tried.

If they should still work without the BMA223 then that is all I needed to know. That saves me a month of trying to sort out removing it from the source code.

It is a pity really. I see no point in buying more if I cannot find why these failed and for the few minutes that they worked I was quite impressed.

**Reply by Ralim** [reply]
Looking at your images it looks like you have a quite long ground loop for measuring the voltage spikes, which I don't think should be a problem, can I have not been about to reproduce this with any supplies I have access to. I have been able to _possibly_ (not reproducibly) get a 3.4 ish spike but nothing at this order of magnitude.

As much as it sucks, it _sounds like_ it could be something odd with your setup as this is so far the first instance of this that I have heard of (though I don't read the main pine64 chats). 

I have seen a somewhat similar issue of a friend killing a ts100 with a poor quality power supply but it was far easier to prove.

Also I find it odd that the main mcu is both surviving the 30V spike and also not clamping that with the ESD diodes.


Can confirm a dead BMA or FUSB won't cause it to totally stop working unless they lockup the i2c bus. If you have removed and cleaned the BMA footprints then that rules that out really.

The OLED screens are somewhat ESD sensitive.


Have you tried powering it up and pressing the front button to see if it heats?

You may have just killed the OLED?

**Reply by Mjolinor** [reply]
Yes. I can get DFU mode OK and dfu_util successfully put the firmware on. No OLED and no heating. The iron draws about 2 ma constant with 10 ma every 20 seconds or so for half a second. I was about to go and read what button press I needed to make it come on when I saw your reply. The "+" button does not seem to affect anything, the "-" button makes the 10 ma into 5 ma every 20 seconds.

**Reply by Mjolinor** [reply]
The pulsing 10ma seems to be the iron checking that there is a tip installed. If I remove the tip the current draw is a  constant 2 ma.

Is the I2C anywhere on the PCB opposite side to the OLED? I would rather not pull it all apart to scope the I2C bus with a tip in. I think I have some new OLEDs of this type somewhere but I may just remove it to see if that makes the iron work.

**Reply by Mjolinor** [reply]
Brain fart. Ignore that, the BMA223 is on that side. D'oh
